File: sdk/js/node/src/emailTemplates.ts
import type { HttpClient } from "./client.js";

export type EmailTemplateType =
  | "welcome"
  | "verify-email"
  | "reset-password"
  | "magic-link"
  | "invite"
  | "mfa-code";

export interface EmailTemplate {
  tenant_id: string;
  type: EmailTemplateType;
  subject: string;
  html_body: string;
  text_body?: string;
  from_name?: string;
  from_address?: string;
  updated_at?: string;
}

export interface UpdateEmailTemplateInput {
  subject?: string;
  html_body?: string;
  text_body?: string;
  from_name?: string;
  from_address?: string;
}

export class EmailTemplates {
  constructor(private readonly http: HttpClient) {}

  get(tenantId: string, type: EmailTemplateType): Promise<EmailTemplate> {
    return this.http.get<EmailTemplate>(
      `/v1/tenants/${encodeURIComponent(tenantId)}/email-templates/${encodeURIComponent(type)}`,
    );
  }

  update(
    tenantId: string,
    type: EmailTemplateType,
    input: UpdateEmailTemplateInput,
  ): Promise<EmailTemplate> {
    return this.http.request<EmailTemplate>(
      "PUT",
      `/v1/tenants/${encodeURIComponent(tenantId)}/email-templates/${encodeURIComponent(type)}`,
      { body: input },
    );
  }

  preview(tenantId: string, type: EmailTemplateType, to: string): Promise<{ sent: boolean }> {
    return this.http.post<{ sent: boolean }>(
      `/v1/tenants/${encodeURIComponent(tenantId)}/email-templates/${encodeURIComponent(type)}/preview`,
      { to },
    );
  }
}

File: sdk/go/emailtemplates.go
package qeetid

import (
	"context"
	"net/http"
	"net/url"
)

type EmailTemplate struct {
	TenantID    string `json:"tenant_id"`
	Type        string `json:"type"`
	Subject     string `json:"subject"`
	HTMLBody    string `json:"html_body"`
	TextBody    string `json:"text_body,omitempty"`
	FromName    string `json:"from_name,omitempty"`
	FromAddress string `json:"from_address,omitempty"`
	UpdatedAt   string `json:"updated_at,omitempty"`
}

type UpdateEmailTemplateInput struct {
	Subject     *string `json:"subject,omitempty"`
	HTMLBody    *string `json:"html_body,omitempty"`
	TextBody    *string `json:"text_body,omitempty"`
	FromName    *string `json:"from_name,omitempty"`
	FromAddress *string `json:"from_address,omitempty"`
}

type EmailPreviewResult struct {
	Sent bool `json:"sent"`
}

type EmailTemplates struct{ http *httpClient }

func (r *EmailTemplates) Get(ctx context.Context, tenantID, templateType string) (*EmailTemplate, error) {
	var out EmailTemplate
	err := r.http.do(ctx, http.MethodGet, "/v1/tenants/"+url.PathEscape(tenantID)+"/email-templates/"+url.PathEscape(templateType), nil, nil, &out, true)
	return &out, err
}

func (r *EmailTemplates) Update(ctx context.Context, tenantID, templateType string, in UpdateEmailTemplateInput) (*EmailTemplate, error) {
	var out EmailTemplate
	err := r.http.do(ctx, http.MethodPut, "/v1/tenants/"+url.PathEscape(tenantID)+"/email-templates/"+url.PathEscape(templateType), nil, in, &out, false)
	return &out, err
}

func (r *EmailTemplates) Preview(ctx context.Context, tenantID, templateType, to string) (*EmailPreviewResult, error) {
	var out EmailPreviewResult
	body := map[string]string{"to": to}
	err := r.http.do(ctx, http.MethodPost, "/v1/tenants/"+url.PathEscape(tenantID)+"/email-templates/"+url.PathEscape(templateType)+"/preview", nil, body, &out, false)
	return &out, err
}
